usaspending_api/reporting/v2/views/agencies/publish_dates.py

```diff
@@ -0,0 +1,147 @@
+import re
+from builtins import next, enumerate
+from copy import deepcopy
+
+from django.contrib.postgres.aggregates import ArrayAgg
+from django.db.models import OuterRef, Subquery, TextField, F, Value, Func, DecimalField, Q
+from django.db.models.functions import Cast
+from django.utils.functional import cached_property
+from jsonpickle import json
+from rest_framework.response import Response
+
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, PaginationMixin
+
+from usaspending_api.common.data_classes import Pagination
+from usaspending_api.common.exceptions import UnprocessableEntityException
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.common.helpers.orm_helpers import ConcatAll
+from usaspending_api.common.validator import customize_pagination_with_sort_columns, TinyShield
+from usaspending_api.references.models import ToptierAgency
+from usaspending_api.reporting.models import ReportingAgencyOverview
+from usaspending_api.submissions.models import SubmissionAttributes
+
+
+class PublishDates(AgencyBase, PaginationMixin):
+    """Returns list of agency submission information, included published and certified dates for the fiscal year"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/reporting/agencies/publish_dates.md"
+
+    def validate_publication_sort(self, sort_key):
+        regex_string = r"publication_date,([2-9]|1[0-2])"
+        if not re.match(regex_string, sort_key):
+            raise UnprocessableEntityException(
+                "publication_date sort param must be in the format 'publication_date,<fiscal_period>'"
+            )
+
+    def get_agency_data(self):
+        agency_filters = [Q(toptier_code=OuterRef("toptier_code"))]
+        if self.filter is not None:
+            agency_filters.append(Q(name__icontains=self.filter))
+        results = (
+            SubmissionAttributes.objects.filter(reporting_fiscal_year=self.fiscal_year)
```

Comment:
This appears to only be obtaining agencies based on the submissions for the specific year (it isn't even filtering out published submissions before a submission window closes which should be an implicit expectation for all account data being returning by the API at this point). See comments in the Jira ticket for a little more detail, specifically https://federal-spending-transparency.atlassian.net/browse/DEV-6439?focusedCommentId=63809

Reply:
yup, this should be addressed now. I've switched the query to use ToptierAgencies.objects.account_agencies, so it'll return all agencies with account data for any fiscal year, as well as all periods w/ blanks if there is no submission for that period.